### PR TITLE
Split crash reproducer data

### PR DESCRIPTION
**Pull request by bertschneider**
If the recorded `FuzzedDataProvider` invocation data gets too long it can not be assigned to one `String` variable in the crash reproducer template anymore. More concretely one constant pool `CONSTANT_Utf8_info` entry can only hold data of
a size up to `uint16`.
This PR splits up the recorded data into multiple chunks to always generate valid crash reproducers.

Furthermore, a `tests` directory is introduced to hold an explicit Jazzer integration test to verify this issue (#269). 
The integration test compiles and executes the generated crash reproducer to verify it's functionality.

### CODE REVIEW

**Comment by fmeum** [comment]
@bertschneider `dd` appears to have a different syntax on macOS (see the pipeline). Maybe there is a common subset of the `dd` arguments among all platforms that we could use here.

File: bazel/FuzzTargetTestWrapper.java [hunk 1]
```diff
@@ -15,32 +15,47 @@
...
     } catch (IOException | ArrayIndexOutOfBoundsException e) {
       e.printStackTrace();
       System.exit(1);
       return;
     }
 
-    ProcessBuilder processBuilder = new ProcessBuilder();
-    Map<String, String> environment = processBuilder.environment();
-    // Ensure that Jazzer can find its runfiles.
-    environment.putAll(runfiles.getEnvVars());
```
Comment (fmeum) [inline bazel/FuzzTargetTestWrapper.java:42]: We should set these env vars in the ProcessBuilder created below.

Comment (bertschneider) [reply bazel/FuzzTargetTestWrapper.java:42]: The env vars didn't look Jazzer specific and all tests passed without. 
I can revert this but wonder which ones Jazzer needs.

**Comment by bertschneider** [comment]
> @bertschneider `dd` appears to have a different syntax on macOS (see the pipeline). Maybe there is a common subset of the `dd` arguments among all platforms that we could use here.

AFAIK `dd` is not available on Windows, so I checked in an appropriate input file. It's not ideal but in this case the file is small enough and should not blow up the repository too much.

File: bazel/FuzzTargetTestWrapper.java (hunk 1 above)
Comment (fmeum) [reply bazel/FuzzTargetTestWrapper.java]: They are not Jazzer-specific, but encode information about how Jazzer can find it's data dependencies provided by Bazel when invoked by the test wrapper. This is always easy to do from the top-level target executed by Bazel, but subprocesses generally have to be passed these variable to find their runfiles in all scenarios (that is, even when executed with bazel run on Windows). If the variables are not passed, runfile discovery will be non-deterministic on Windows.

Comment (bertschneider) [reply bazel/FuzzTargetTestWrapper.java]: Done. Thx for the explanation.

**Comment by bertschneider** [comment]
Added a mapping of all parameters not starting with `-` to their actual locations using `rlocation`. With this change all tests are successful.

**Comment by fmeum** [comment]
Thanks, this looks pretty good now! 

One last nit: Could we drop the `Example` prefix from the files and targets under `test`? The names get pretty long and now that this doesn't live under `examples` any more, the prefix has lost its purpose.

**Comment by bertschneider** [comment]
Ah, yes, good point. I removed the `Example` prefix. Thank you very much for the review(s)!

**Review by fmeum — APPROVED** [review]
Thanks! Feel free to merge this now (probably squash merge).

**Pull request merged**